File: challenges/2026-08-30-loyalty-tier-lookup/solution.ts
function findTier(thresholds: number[], points: number): number {
  // TODO: implement
  return -1;
}

type Case = [number[], number, number];

const tests: Case[] = [
  [[100, 250, 500, 1000], 300, 1],
  [[100, 250, 500, 1000], 50, -1],
  [[100, 250, 500, 1000], 1000, 3],
  [[100, 250, 500, 1000], 249, 0], // extra: just below the next threshold
  [[5], 5, 0], // extra: single-tier array, exact match
];

for (const [thresholds, points, expected] of tests) {
  const got = findTier(thresholds, points);
  const status = got === expected ? "PASS" : "FAIL";
  console.log(
    `${status} thresholds=${JSON.stringify(thresholds)} points=${points} expected=${expected} got=${got}`
  );
}

File: challenges/2026-08-30-loyalty-tier-lookup/solution.py
from typing import List


def find_tier(thresholds: List[int], points: int) -> int:
    """Return the highest tier index i with thresholds[i] <= points, or -1."""
    # TODO: implement
    pass


if __name__ == "__main__":
    tests = [
        ([100, 250, 500, 1000], 300, 1),
        ([100, 250, 500, 1000], 50, -1),
        ([100, 250, 500, 1000], 1000, 3),
        ([100, 250, 500, 1000], 249, 0),   # extra: just below the next threshold
        ([5], 5, 0),                        # extra: single-tier array, exact match
    ]

    for thresholds, points, expected in tests:
        got = find_tier(thresholds, points)
        status = "PASS" if got == expected else "FAIL"
        print(f"{status} thresholds={thresholds} points={points} expected={expected} got={got}")
